### PR TITLE
AspectJ

### PR DESCRIPTION
**Pull request by benrudolph**
@wpride did some digging into how we can more cleaning define things that will execute before and after a function. This would be clutch for the locking code. anyways, i think i got it mostly setup, i think i'm just missing some sort of wiring. here's the documentation i'm working off of: http://docs.spring.io/spring/docs/current/spring-framework-reference/html/aop.html

curious if you have any thoughts/have done this before

### CODE REVIEW

File: src/main/java/aspects/LockOnUsername.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import org.springframework.stereotype.Component;
+
+/**
+ * Created by benrudolph on 10/19/16.
+ */
+@Component
+@Aspect
+public class LockOnUsername {
+
+    @Around("@annotation(annotations.LockAndBlock)")
```
Comment (benrudolph) [inline src/main/java/aspects/LockOnUsername.java]: this should run around any method that has the `@LockAndBlock` annotation, but it's not currently working

File: src/main/java/aspects/LockOnUsername.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    public Object aroundLockAndBlockOnUsername(ProceedingJoinPoint pjp) throws Throwable {
+        // start stopwatch
+        if (1== 1) {
+            throw new RuntimeException("BEN");
```
Comment (benrudolph) [inline src/main/java/aspects/LockOnUsername.java]: (just for testing purposes)

**Comment by snopoke** [comment]
I've used AOP before and it's very useful for this kind of thing. Don't think I'll be much help with setting it up though.

**Comment by benrudolph** [comment]
Ah that's good to hear. It does look incredibly useful once the setup is done. will offline with will about it

**Comment by wpride** [comment]
Awesome find, wasn't aware of this feature of Spring at all. Would be interested on pairing on this if you're looking for an assist

**Comment by snopoke** [comment]
This isn't specific to spring. You can use it in any Java runtime.

**Pull request closed without merging**